app/marketing/management/commands/send_tax_report.py

```diff
@@ -0,0 +1,347 @@
+                if key in data_dict.keys():
+                    annotation.update(pdfrw.PdfDict(V='{}'.format(data_dict[key])))
+    pdfrw.PdfWriter().write(os.path.join(recipient_path, template_path[1]) , template_pdf)
+
+
+def get_profile_location(profile):
+    us_worker = False
+    location = NO_LOCATION
+    location_temp = ''
+    if profile.location:
+        location_components = json.loads(profile.location)
+        if LOCALITY in location_components:
+            location_temp += location_components[LOCALITY] 
+        if COUNTRY in location_components:
+            country = location_components[COUNTRY]
+            if location_temp:
+                location_temp += ', ' + country
+            else:
+                location_temp += country
+        if CODE in location_components:
+            code = location_components[CODE]
+            if location_temp:
+                location_temp += ', ' + code
+            else:
+                location_temp += code
+            # check if the user is from US
+            if code == US:
+                us_worker = True
+    elif profile.locations:
+        location_components = profile.locations[-1]
+        if CITY in location_components:
+            location_temp += location_components[CITY]
+        if COUNTRY_NAME in location_components:
+            country_name = location_components[COUNTRY_NAME]
+            if location_temp:
+                location_temp += ', ' + country_name
+            else:
+                location_temp += country_name
+        if COUNTRY_CODE in location_components:
+            country_code = location_components[COUNTRY_CODE]
+            if location_temp:
+                location_temp += ', ' + country_code
+            else:
+                location_temp += country_code
+            if location_components[COUNTRY_CODE] == US:
+                us_worker = True
+    if location_temp:
+        location = location_temp
+    return location, us_worker
+
+
+def zip_dir(username, username_path):
+    zip_file_path = os.path.join(username_path, username) + '.zip'
+    zipf = zipfile.ZipFile(zip_file_path, 'w', zipfile.ZIP_DEFLATED)
+    for root, dirs, files in os.walk(username_path):
+        for filename in files:
+            if filename.endswith('.zip'):
+                continue
+            abs_path = os.path.join(root, filename)
+            zipf.write(abs_path, abs_path.split('code/tax_report')[1])
+    zipf.close()
+    return zip_file_path
+
+
+class Command(BaseCommand):
+
+    help = 'the tax report for last year'
+
+    def handle(self, *args, **options):
+        to_emails = []
+        zip_paths = []
+        profiles = Profile.objects.all()
+        tax_path = os.path.join(os.getcwd(), TAX_REPORT_PATH)
+        if not tax_path:
+            os.makedirs(tax_path)
+        for p in profiles:
+            csv_record = []
+            us_workers = {}
+            # Bounties
+            for b in p.get_sent_bounties:
```

Comment:
are we still planning on using `Earnings` instead?!?

Reply:
Hi kevin, if you want i can change it into the code without problems but it requires a little more time to make some tests. I could do it during next week. 
One question: In `Earning` the `source_id` field correspond to `id` in `BountyFulfillment`, for instance in bounty case ? Because i try to compare `Earning` `source_id` -> `BountyFulfillment` `bounty_id` -> `Bounty` `id` and for instance the `value_usd` does not seem to match between earning and bounty. Could you check it ? thanks

Reply:
Done